proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -690,13 +690,17 @@ void teardown() {
       connectedServer.disconnect();
     }
 
-    Optional<Player> connectedPlayer = server.getPlayer(this.getUniqueId());
     server.unregisterConnection(this);
 
     DisconnectEvent.LoginStatus status;
-    if (connectedPlayer.isPresent()) {
-      status = connectedPlayer.get() == this ? LoginStatus.SUCCESSFUL_LOGIN
-          : LoginStatus.CONFLICTING_LOGIN;
+    if (server.getPlayer(this.getUniqueId()).isPresent()) {
+      Player connectedPlayer = server.getPlayer(this.getUniqueId()).get();
```

Comment:
Arguably this is worse than before (even if player lookups are constant). I disagree with @MrIvanPlays, previous code was fine.

Reply:
I'll just revert it then

Reply:
This revised code actually introduces a bug.

Reply:
Yeah, figured such would occur. Will do as i stated previously